### PR TITLE
hyprnotify: init at 0.7.0

### PR DESCRIPTION
**Pull request by Bloeckchengrafik**
## Description of changes
This PR adds the package [hyprnotify](https://github.com/codelif/hyprnotify). From the README:
Hyprnotify is a [Freedesktop.org](https://specifications.freedesktop.org/notification-spec/notification-spec-latest.html) compliant notification daemon implementing hyprctl notify as its backend.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md). _i think so at least_

---

As this is my first contribution to nixpkgs, please tell me if i did something wrong! Thanks!

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hy/hyprnotify/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  pkgs,
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprnotify/package.nix:3]: `pkgs` is not allowed here. `callPackage` will expand all attributes in `pkgs` and pass the arguments needed by this function.

For example, if you need `buildGoModule`, you just write `buildGoModule` here. (side note: It's a good idea to match the order in which these arguments are used. However, we usually put `lib` first.)

File: pkgs/by-name/hy/hyprnotify/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    repo = "hyprnotify";
+    rev = "v${version}";
+    hash = "sha256-dL+W+iMwRNw9042bs2XUFPMCCqIvDENXOMzhcLh+RL4=";
+  };
+
+  nativeBuildInputs = with pkgs; [ pkg-config ];
+  buildInputs = with pkgs; [ alsa-lib ];
+  vendorHash = "sha256-AZDtaiSNq7em876Q9f+YeDxboqVwA8IE9dDM6zggFXs=";
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprnotify/package.nix:22]: `with lib` is not such a good idea. We're already trying to reduce it as much as possible.

File: pkgs/by-name/hy/hyprnotify/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  version = "0.7.0";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "codelif";
+    repo = "hyprnotify";
+    rev = "v${version}";
+    hash = "sha256-dL+W+iMwRNw9042bs2XUFPMCCqIvDENXOMzhcLh+RL4=";
+  };
+
+  nativeBuildInputs = with pkgs; [ pkg-config ];
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprnotify/package.nix:18]:
```suggestion
  nativeBuildInputs = with pkgs; [ pkg-config ];

```

File: pkgs/by-name/hy/hyprnotify/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  buildInputs = with pkgs; [ alsa-lib ];
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprnotify/package.nix:19]:
```suggestion
  buildInputs = with pkgs; [ alsa-lib ];

```

File: pkgs/by-name/hy/hyprnotify/package.nix (hunk 1 above)
Comment (Bloeckchengrafik) [reply pkgs/by-name/hy/hyprnotify/package.nix]: i think my recent commit fixes this, please LMK if there is anything else!

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**